game connection mvp

Govorunb opened the pull request:
closes #14 somewhat

Govorunb committed:
game connection mvp

Govorunb committed:
Merge branch 'main' into game

Govorunb committed:
make manual calls easier

Alexejhero commented on ebs/package.json:
```diff
@@ -8,10 +8,12 @@
     },
     "dependencies": {
         "@twurple/ebs-helper": "^7.1.0",
+        "@types/express-ws": "^3.0.4",
```
this should be installed with -D

Alexejhero commented on ebs/src/index.ts:
```diff
@@ -33,18 +34,25 @@ async function main() {
             });
             break;
         } catch {
+            if (!process.env.DB_HOST) {
```
renamed to MYSQL_HOST

Alexejhero commented on ebs/src/modules/game.ts:
move to modules/game/index.ts at the bare minimum

Alexejhero commented on ebs/src/modules/game/connection.ts:
```diff
@@ -0,0 +1,113 @@
+        }
+        this.socket = ws;
+        ws.on('connection', () => {
+            console.log(`Connected`);
```
connected to what?

Alexejhero commented on ebs/src/modules/game/connection.ts:
```diff
@@ -0,0 +1,113 @@
+                this.handshake = true;
+                const reply = {
+                    ...this.makeMessage(MessageType.HelloBack),
+                    allowed: (msg as HelloMessage).version == VERSION,
```
this unnecessary can be fixed with proper typescript

Alexejhero commented on ebs/src/modules/game/messages.server.ts:
```diff
@@ -0,0 +1,27 @@
+
+export enum CommandInvocationSource {
+    Swarm,
+    Dev
```
rename to Backseat

Govorunb replied:
no

Alexejhero commented on ebs/src/modules/game/messages.ts:
```diff
@@ -0,0 +1,33 @@
+export type TwitchUser = {
+    /** Channel id */
+    id: number,
+    /** Twitch username (login name) */
+    userName?: string,
+    /** User's chosen display name. */
+    displayName?: string
```
This should never be null, right?

Alexejhero commented on ebs/src/modules/game/messages.server.ts:
```diff
@@ -0,0 +1,27 @@
+
+export type ServerMessage = Message & {
+    /** User that triggered the message. e.g. for redeems, the user who bought the redeem. */
+    user?: TwitchUser | null
```
Pick one, not both

```suggestion
    user: TwitchUser | null
```

```suggestion
    user?: TwitchUser
```

Govorunb committed:
physically forced to push unfinished work, where is the union

Govorunb committed:
Merge branch 'main' into game

Govorunb committed:
Merge branch 'main' into game

Govorunb committed:
fix renamed var

Govorunb committed:
more viable product

address redeems by id
choice to announce redeem
actually send the redeem to the game

Govorunb committed:
announce stuff

fix build error

Govorunb committed:
Merge branch 'main' into game

Govorunb committed:
classic off by one error

Govorunb committed:
Merge branch 'main' into game
